Cargo.toml

```diff
@@ -18,3 +18,11 @@ appveyor = { repository = "YaLTeR/circular-queue" }
 
 [build-dependencies]
 version_check = "0.9"
+
+[features]
+default = []
+serde_support = ["serde", "serde_test"]
+
+[dependencies]
+serde = {version = "1.0", optional = true}
+serde_test = {version = "1.0", optional = true}
```

Comment:
I think serde_test can go into dev-dependencies since it's only used for tests?

Reply:
I tried that - unfortuantely, dev-dependencies cannot be optional, apparently.  So I'm not sure how to keep serde behind a feature but keep serde_test optional.

[https://github.com/rust-lang/cargo/issues/1596](url)

Reply:
Okay, looks like this is the best way to do it currently indeed then.

Reply:
I added a couple of additional dependencies to test JSON and Bincode serialization.  I put these behind a different feature (including the `serde_test` crate).  So I think that means if you just want `serde_support`, you do not get any additional dependencies.  But to test, you need `serde_test, serde_json, bincode`.  To me, that seems pretty reasonable.  If you prefer something different, LMK.

Reply:
It is not required to made `serde_test` optional in dev-dependencies. It will be used only if some your code will depend use it, and this code will be included only when you enable the `serde_support` feature during tests. So if you run
```console
cargo test --no-default-features
```
the `serde_test` dependency will not compiled (and even downloaded if I remember correctly) because you have no code that uses it.